Versioning_oM/Attributes/UpgraderAttribute.cs

```diff
@@ -0,0 +1,66 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2025, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Base;
+using BH.oM.Base.Debugging;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+
+namespace BH.oM.Versioning
+{
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = false)]
+    [Description("Attribute identifying a class as an upgrader for versioning.")]
```

Comment:
Might be worth adding something here specifying that the version numbers should be the version you are upgrading to and not from. That if you are working in 8.1, this should be 8.1, not 8.0.

```suggestion
    [Description("Attribute identifying a class as an upgrader for versioning. The version specified should be the the version that the object is being upgraded to which generally correspond to the current version of the BHoM.")]
```

Not sure if my suggested change is clear enough.